openapi-cli/src/test/java/io/ballerina/openapi/CodeGeneratorTest.java

```diff
@@ -183,6 +183,30 @@ public void generateTypesWithNullableFields() {
         }
     }
 
+    @Test(description = "Test Ballerina types generation when referred by another record with no additional fields")
```

Comment:
Since the fix is related to schema shall we add a schema specific test like [ReferenceResolveTests](https://github.com/ballerina-platform/ballerina-openapi/blob/master/openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/ReferenceResolveTests.java) ? CodeGenerator tests are added when there is a change in the high level code generation process.